docs/developer-guide/working-with-plugins.md

```diff
@@ -18,6 +18,21 @@ module.exports = {
 };
 ```
 
+### Preprocessors in Plugins
+
+You can also create plugins that would tell ESLint how to process files other then JavaScript. In order to create a
+preprocessor, object that is exported from your module has to conform to the following interface:
+
+```js
+{
+    preprocess: function(string) { return [string] } //Takes text of the file and returns array of texts to be processed
+    postprocess: function([[Message]]) { return [Message] //Takes array of array of error messages, one for each text
```

Comment:
That means I will need to parse string manually in order to figure out which extensions map to which preprocessor. How about making key arbitrary name and adding extensions array to the object? Something like this:
```js
module.exports = {
    ...
    processors: {
        "html": {
            preprocess: function() {},
            postprocess: function() {},
            extensions: ['html', 'htm']
        }
    }
};
```
I don't really like the idea of arbitrary name, but the only other way I can think of is making processors into array, not an object.